packages/server/lib/controllers/scripts/config/getScriptsConfigOpenAI.ts

```diff
@@ -0,0 +1,212 @@
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { providerNameSchema } from '../../../helpers/validation.js';
+import type { StandardNangoConfig, NangoModelField } from '@nangohq/types';
+import { getSyncConfigsAsStandardConfig } from '@nangohq/shared';
+
+export const validationParams = z
+    .object({
+        provider: providerNameSchema
+    })
+    .strict();
+
+interface OpenAIFunction {
+    name: string;
+    description: string;
+    parameters: {
+        type: string;
+        properties: Record<string, any>;
+        required: string[];
+    };
+}
+
+interface LegacyModelField {
+    name: string;
+    type: string;
+    description?: string;
+    optional?: boolean;
+}
+
+function isLegacyModelField(field: NangoModelField | LegacyModelField): field is LegacyModelField {
+    return 'type' in field;
+}
+
+function getFieldType(field: NangoModelField | LegacyModelField): string {
+    if (isLegacyModelField(field)) {
+        return field.type;
+    }
+
+    // Handle array fields
+    if ('array' in field && field.array) {
+        return 'array';
+    }
+
+    if (typeof field.value === 'string') {
+        return field.value;
+    }
+    if (typeof field.value === 'number') {
+        return 'number';
+    }
+    if (typeof field.value === 'boolean') {
+        return 'boolean';
+    }
+    if (field.value === null) {
+        return 'null';
+    }
+    if (Array.isArray(field.value)) {
+        return 'array';
+    }
+    return 'string'; // Default to string if unknown
+}
+
+function getFieldDescription(field: NangoModelField | LegacyModelField): string {
+    if (isLegacyModelField(field)) {
+        return field.description || '';
+    }
+
+    // Try to parse description from the field value if it's a string
+    if (typeof field.value === 'string') {
+        // Look for a description in the string value
+        const descriptionMatch = field.value.match(/\/\*\*(.*?)\*\//);
+        if (descriptionMatch && descriptionMatch[1]) {
+            return descriptionMatch[1].trim();
+        }
+    }
+
+    return '';
+}
+
+function isFieldOptional(field: NangoModelField | LegacyModelField): boolean {
+    if (isLegacyModelField(field)) {
+        return field.optional || false;
+    }
+    return field.optional || false;
+}
+
+function parseParameterDescriptions(description: string): Record<string, string> {
+    const descriptions: Record<string, string> = {};
+    const lines = description.split('\n');
+
+    for (const line of lines) {
+        // Match markdown list items that describe parameters
+        const match = line.match(/^-\s*(\w+):\s*(.+)$/);
+        if (match && match[1] && match[2]) {
+            const [, paramName, paramDesc] = match;
+            descriptions[paramName] = paramDesc.trim();
+        }
+    }
+
+    return descriptions;
+}
+
+function transformToOpenAIFunctions(configs: StandardNangoConfig[]): OpenAIFunction[] {
```

Comment:
it's not possible to use jsonSchema? It would be simpler

Reply:
Not entirely, it is a subset of the JSON schema so can't just plug and play unfortunately

Reply:
I meant instead of using NangoModelField that are very custom and might be gone sometimes